realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -316,6 +316,11 @@ private static void initializeRealm(Realm realm) {
                 columnInfoMap.put(modelClass, mediator.validateTable(modelClass, realm.sharedGroupManager.getTransaction()));
             }
             realm.schema.columnIndices = new ColumnIndices(columnInfoMap);
+
+            final Transaction transaction = realm.getConfiguration().getInitialDataTransaction();
```

Comment:
I don't think this is enough to ensure that we only trigger this when the file is first created, e.g. I am pretty sure this will get triggered in the following case:

```
RealmConfiguration config = new RealmConfiguration.Builder(context).build();
Realm.getInstance(config).close();
Realm.getInstance(config).close(); // This will trigger it again since the schema has been cleared
```

Also, the file might be copied from assets instead. I think the only way of doing this is to have a method that checks for the existence of the Realm file and setting a flag. A good place would probably be just before calling the BaseRealm constructor.

Also, unit tests are needed to verify the behaviour of this.